File: all-submissions/0102-binary-tree-level-order-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        List<List<Integer>> answer = new ArrayList<List<Integer>>();
        if (root == null) return answer;
        // if (root.left == null && root.right == null) return answer;
        Queue<TreeNode> queue = new LinkedList<TreeNode>();

        int level = 0;
        queue.add(root);
        while (!queue.isEmpty()){
            answer.add(new ArrayList<Integer>());
            int level_length = queue.size();
            for (int i=0; i< level_length; i++) {
                TreeNode node = queue.remove();
                answer.get(level).add(node.val);
                if (node.left != null) queue.add(node.left);
                if (node.right != null) queue.add(node.right);
            }
            level++;
        }

        return answer;

        
    }
}

File: all-submissions/0102-binary-tree-level-order-traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        queue = deque()
        if not root: return res
        node = root
        queue.append(node)
        while queue:
            ans = []
            nodesInCurrLevel = len(queue)
            for i in range(nodesInCurrLevel):
                curr_node = queue.popleft()
                ans.append(curr_node.val)
                if curr_node.left:
                    queue.append(curr_node.left)
                if curr_node.right:
                    queue.append(curr_node.right)
            res.append(ans)
        return res
